src/components/tx/security/tenderly/index.tsx

```diff
@@ -45,7 +51,7 @@ const TxSimulationBlock = ({ transactions, disabled, gasLimit }: TxSimulationPro
 
     simulateTransaction({
       safe,
-      executionOwner: wallet.address,
+      executionOwner: isRecovery ? safe.owners[0].value : wallet.address,
```

Comment:
Maybe it's better to pass `executionOwner` instead of `isRecovery`?

Reply:
The "guardian" (`wallet.address`) is unlikely an owner of the Safe so we have to explicitly pass one. The simulation would otherwise fail.

Reply:
I meant the prop of `TxSimulationBlock`. Instead of passing `isRecovery` which is too specific, lift that logic to the caller and pass an optional `executionOwner` instead, with a fallback to `wallet.address`.

Reply:
I commited the suggestion directly to the epic branch in ba3b7064f2367e95a62088b6b423792be0f6e606.